packages/ui/src/components/styled-link.tsx

```diff
@@ -1,27 +1,86 @@
+import { MouseEvent, RefAttributes } from 'react'
 import type { LinkProps } from 'react-router-dom'
 
 import { useRouterContext } from '@/context'
 import { cn } from '@utils/cn'
 import { cva, VariantProps } from 'class-variance-authority'
 
-const linkVariants = cva(
-  'whitespace-nowrap decoration-transparent decoration-1 underline-offset-4 transition-colors duration-200',
-  {
-    variants: {
-      variant: {
-        default: 'hover:decoration-foreground-1 text-cn-foreground-1',
-        secondary: 'hover:decoration-foreground-4 text-cn-foreground-2',
-        accent: 'hover:decoration-foreground-accent text-cn-foreground-accent'
-      }
+import { Icon, IconProps } from './icon'
+
+export const linkVariants = cva('cn-link', {
+  variants: {
+    variant: {
+      default: 'cn-link-default',
+      secondary: 'cn-link-secondary'
+    },
+    size: {
+      default: '',
+      sm: 'cn-link-sm'
     }
   }
-)
+})
 
-export type StyledLinkProps = LinkProps & RefAttributes<HTMLAnchorElement> & VariantProps<typeof linkVariants>
+interface StyledLinkProps extends LinkProps, RefAttributes<HTMLAnchorElement>, VariantProps<typeof linkVariants> {
+  /**
+   * If true, the 'chevron-left' icon will be displayed before the link text.
+   * If a string, that string will be used as the icon name.
+   */
+  prefixIcon?: boolean | IconProps['name']
+  /**
+   * If true, the 'arrow-to-top-right' icon will be displayed before the link text.
+   * If a string, that string will be used as the icon name.
+   */
+  suffixIcon?: boolean | IconProps['name']
+  /**
+   * If true, the link will be disabled and not clickable.
+   * The default value is false.
+   */
+  disabled?: boolean
+}
 
-export const StyledLink = ({ className, variant = 'default', ...props }: StyledLinkProps) => {
+const StyledLink = ({
+  className,
+  variant = 'default',
+  children,
+  prefixIcon,
+  suffixIcon,
+  disabled = false,
+  size,
+  onClick,
+  ...props
+}: StyledLinkProps) => {
   const { Link } = useRouterContext()
 
-  return <Link className={cn(linkVariants({ variant }), className)} {...props} />
+  const handleClick = (e: MouseEvent<HTMLAnchorElement>) => {
+    if (disabled) {
+      e.preventDefault()
+      e.stopPropagation()
+    } else {
+      onClick?.(e)
+    }
+  }
+
+  return (
+    <Link
+      {...props}
+      className={cn(linkVariants({ variant, size }), className)}
+      onClick={handleClick}
+      data-disabled={disabled}
+      aria-disabled={disabled}
+    >
+      {!!prefixIcon && (
+        <Icon className="cn-link-icon" name={typeof prefixIcon === 'string' ? prefixIcon : 'chevron-left'} skipSize />
+      )}
+      {children}
+      {!!suffixIcon && (
+        <Icon
+          className="cn-link-icon"
+          name={typeof suffixIcon === 'string' ? suffixIcon : 'arrow-to-top-right'}
+          skipSize
+        />
+      )}
+    </Link>
+  )
 }
+
+export { StyledLink as Link, type StyledLinkProps as LinkProps }
```

Comment:
Instead of using `as X`, can we just change the component and type names directly? We can alias things coming out of react-router-dom and `useRouterContext` instead

Reply:
Fixed